File: Src/PHP/personClient.php
<?php

namespace IDV;
error_reporting(E_ALL);

require_once __DIR__ . '/thrift/Thrift/ClassLoader/ThriftClassLoader.php';
require_once __DIR__ . '/thrift/IDV/PersonService.php';
require_once __DIR__ . '/thrift/IDV/Types.php';

use Thrift\ClassLoader\ThriftClassLoader;
use Thrift\Protocol\TBinaryProtocol;
use Thrift\Transport\TSocket;
use Thrift\Transport\TBufferedTransport;

// Load
$loader = new ThriftClassLoader();
$loader->registerNamespace('Thrift', __DIR__ . '/thrift');
$loader->registerDefinition('IDV', __DIR__ . '/thrift/IDV');
$loader->register();

// Init
$server = '127.0.0.1';
$port = 8989;
$socket = new TSocket($server, $port);
$transport = new TBufferedTransport($socket, 1024, 1024);
$protocol = new TBinaryProtocol($transport);
$client = new PersonServiceClient($protocol);

// Config
$socket->setSendTimeout(30 * 1000);
$socket->setRecvTimeout(30 * 1000);

// Connect
$transport->open();

// Call...
$response = $client->getPersonById(125);

// Print response...
var_dump($response);

// Close
$transport->close();

File: Src/Java/src/PersonClient.java
import java.lang.System;

import org.apache.thrift.*;
import org.apache.thrift.protocol.*;
import org.apache.thrift.transport.*;

import IDV.*;

public class PersonClient {
    private static final int PORT = 8989;

    public static void main(String[] args) throws TException {
       TTransport transport = new TSocket("127.0.0.1", PORT);
       TProtocol protocol = new TBinaryProtocol(transport);
       PersonService.Client client = new PersonService.Client(protocol);
       long start = 0;
       
       transport.open();

       System.out.println("開始呼叫...");
       start = System.currentTimeMillis();   // store/remember start time
       Person person = client.getPersonById(125);

       System.out.println("呼叫结果：" + person.toString());
       transport.close();

       System.out.println("消耗時間：" + (System.currentTimeMillis()-start) + "ms");  // calculate the elapsed time.
    }
}
